packages/xo-web/src/common/iso-device.js

```diff
@@ -70,13 +86,17 @@ export default class IsoDevice extends Component {
   _showWarning = () => alert(_('cdDriveNotInstalled'), _('cdDriveInstallation'))
 
   render () {
-    const { cdDrive, mountedIso } = this.props
+    const { cdDrive, isAdmin, mountedIso } = this.props
+    const resourceSet = this._getResolvedResourceSet()
+    const SelectVdi_ =
+      isAdmin || resourceSet === undefined ? SelectVdi : SelectResourceSetsVdi
 
     return (
       <div className='input-group'>
-        <SelectVdi
-          srPredicate={this._getPredicate()}
+        <SelectVdi_
           onChange={this._handleInsert}
+          resourceSet={isAdmin ? undefined : resourceSet}
```

Comment:
It should be the same test as above (add `|| resourceSet === undefined`) since if the user is not an admin and `resourceSet` is `undefined`, you'll pass a useless `resourceSet` prop.
(you can create a `useResourceSet` variable since you'll test the same thing twice, now)